### DISCUSSION TITLE
Suggestions on improvements for the security page on the website simplelogin website.

### DISCUSSION THREAD

**Discussion by c0nfigurati0n** [simple-login/app#744] [Feature Requests]
# Info
Description: here are some suggestions i recommend that should be made on the security page of simplelogin's website.

# Bio
I recommend that the simplelogin team should updated their security page to include the following.

- Add an explation on how you guys store passwords in detail (e.g. do you guys hash passwords, what hashing algorithm is used, etc).

- Add a detailed explation on how you guys store other information that is more permanent (e.g. email addresses, usernames, profile pictures are stored in what type of format, using what algorithm, etc).

- Mention that simplelogin also offers 2FA to its customers on the security page.

- Add an explation on how you guys protect against DoS/DDoS attacks, so that the deliverability of your guys service(s) doesn't get messed up (and how you respect people's privacy while doing so).

# Additional things to look into

- Adding a [Transparency Report](https://en.wikipedia.org/wiki/Transparency_report).

- Adding a [Warrant Canary](https://en.wikipedia.org/wiki/Warrant_canary).

- **Publishing all of the PGP keys thumb prints**.

- Starting a cryptographic identity for simplelgoin with pgp and making that public everywhere (explation of a [cryptographic identity](https://www.youtube.com/watch?v=oc5844dyrsc)).

- Creating a more in depth ToS ([Terms of Service](https://en.wikipedia.org/wiki/Terms_of_service)).

# Finished/added by the simplelogin team

- Adding the official link to France's SIREN directory/database on the bottom of every page so people can verify simplelogin is a registered business.

- Add a Vulnerability Reporting/[Bug bounty program](https://en.wikipedia.org/wiki/Bug_bounty_program).

- Adding a Mastodon Social Media Account.

- Adding a decided security email.

- Mention how you guys manage the PoLP ([Principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege)) Policy for your infrastructure and internal services.

- Add an explation on how you guys handle domain registration (so that people know a domain doesn't expire on accident or something like that).

**Comment by c0nfigurati0n** [comment]
@nguyenkims just mentioning you so that you see this :D

**Reply by nguyenkims** [reply]
Hey thanks for the request, it seems very complete indeed. We plan to do a security audit this year and several technical points will be covered in this report. We'll make sure to make the report accessible to anyone.

**Reply by c0nfigurati0n** [reply]
@nguyenkims Yeah no problem. I'm just trying to help in anyway i can! Also could you please clarify if this audit it separate from the security page or will it just be updating the security page? Because if this in a third party company doing the audit, then it will be separate from the security page. No?

**Reply by nguyenkims** [reply]
It's going to be separated from the security page. There will be a link to download the report from the security page though for those who want to see it.

**Reply by c0nfigurati0n** [reply]
Cool! Also does this mean that any of the recommendations i made won't be implemented (just asking to make sure).

**Reply by nguyenkims** [reply]
@MrMuffinIsTaken we'll try to implement some recommendations :). 

> Adding the official link to France's SIREN directory/database on the about page so people can verify simplelogin's team members are officially part of simplelogin SAS (if i recall correctly this is possible).

This one isn't available as far as I know.

**Reply by c0nfigurati0n** [reply]
Hey Son, did some digging and found the data base. Here you go: https://www.sirene.fr/sirene/public/recherche, but you're right, you can't like all of them together, but you can look them up if they're doing business

**Reply by nguyenkims** [reply]
Hey this page has been updated to include a link to the official SimpleLogin registration https://simplelogin.io/imprint/

**Reply by c0nfigurati0n** [reply]
Hey Son, alright. I'll updated the discussion to mark that as complete.

**Reply by c0nfigurati0n** [reply]
@nguyenkims anymore update's from the list that i'm not aware of that should be crossed off?

**Reply by nguyenkims** [reply]
@c0nfigurati0n Hey we didn't have the time to look at the list yet since the integration with Proton. Now that SL is running by Proton team, several points on the list should be ticked. For ex, PoLP is already implemented. Domain registration is now handled by a dedicated team. DoS/DDoS attacks are also handled by Proton infra team which has a lot of experiences with these attacks. I don't have all the details about every security measure though.

**Reply by c0nfigurati0n** [reply]
@nguyenkims crossed the ones you mentioned off the list.
